lib/jerboa/client/credentials.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Jerboa.Client.Credentials do
+  @moduledoc false
+  ## Data structure containing client's authentication data
+
+  defstruct [:username, :realm, :secret, :nonce]
+
+  @type t :: %__MODULE__{
```

Comment:
This looks really interesting! I'll try to implement the approach from the video here.